app/controllers/reactions_controller.rb

```diff
@@ -1,5 +1,6 @@
 class ReactionsController < ApplicationController
   before_action :set_cache_control_headers, only: [:index], unless: -> { current_user }
+  before_action :check_limit, only: [:create]
```

Comment:
If you want to save yourself from always having to define an extra method:

```ruby
before_action(only: [:create]) { rate_limit!(:reaction_creation) }
```

I'm not entirely convinced that the `before_action`s really are necessary for most of these cases. They cancel the action if the callback renders or redirects and also cancel additional callbacks that might otherwise run. Our action here does either nothing or raises, so the result is pretty much the same.

But then I think callbacks are generally overused in Rails, especially when they are only applied to single actions.

Reply:
I had methods before and it was suggested I put them in before_actions 🤷 6/half dozen the other I guess.